scripts/deployStakingAndFactory.ts

```diff
@@ -33,13 +44,21 @@ async function main() {
     throw `No config found for network ${config}`;
   }
 
-  const factory = await deployFactory();
-  const staking = await deployStaking(
+  const [owner] = await ethers.getSigners();
+
+  const FactoryFactory = await ethers.getContractFactory("TalentFactory");
+  const factory = (await upgrades.deployProxy(FactoryFactory, [])) as TalentFactory;
+
+  const staking = (await deployContract(owner, StakingArtifact, [
+    dayjs().add(10, "minute").unix(),
+    dayjs().add(40, "year").unix(),
+    ethers.utils.parseUnits("400000000"),
     config.usdStableCoinContract,
     factory.address,
-    parseUnits(config.talPriceInUsd),
-    parseUnits(config.talentPriceInTal)
-  );
+    parseUnits(config.talPriceInUsd), // how much cUSD must be spent for 1 TAL
+    parseUnits(config.talentPriceInTal), // how much TAL must be spent for 1 Talent Token
+  ])) as Staking;
```

Comment:
are you going to deploy a new staking contract? shouldn't you instead be upgrading the existing one?